core/vm/errors.go

```diff
@@ -81,6 +81,11 @@ type ErrInvalidOpCode struct {
 
 func (e *ErrInvalidOpCode) Error() string { return fmt.Sprintf("invalid opcode: %s", e.opcode) }
 
+// NewErrInvalidOpCode is only used in tests to generate an exact ErrInvalidOpCode error.
+func NewErrInvalidOpCode(opcode OpCode) *ErrInvalidOpCode {
```

Comment:
Please make this a private function inside the test file.

Reply:
This comment makes sense. While having tried other methods, it's tricky to initialize an `ErrInvalidOpCode` type outside the `vm` package because the `opcode` field of `ErrInvalidOpCode` is unexported. Thus, this unit test is removed to ensure the code is clean.

Reply:
Added the unit test back in [e26ecee](https://github.com/ethereum/go-ethereum/pull/31198/commits/e26eceee77b1a7b1dec79f1535960bb5981cf46c). Replaced the `!reflect.DeepEqual(err, tc.expectErr)` to error string matching.